File: server/schema/application.py
from uuid import UUID
from pydantic import BaseModel, EmailStr
from datetime import date

from enum import Enum

class StatusType(Enum):
    PENDING = 'PENDING'
    APPROVED = 'APPROVED'
    REJECTED = 'REJECTED'

class LeaveType(Enum):
    CL = 'CL'
    SPL = 'SPL'
    BTR = 'BTR'
    EL = 'EL'

class CreateApplication(BaseModel):
    leave_type: LeaveType
    start_date: date
    end_date: date
    reason: str

    class Config:
        from_attributes = True

File: client/src/types/application.ts
export enum Status{
    PENDING = 'PENDING',
    REJECTED = 'REJECTED',
    APPROVED = 'APPROVED'
};

export enum LeaveType{
    CL = 'CL',
    SPL = 'SPL',
    BTR = 'BTR',
    EL = 'EL'
};

export interface Application{
    id: string;
    applicantId: string;
    leaveType: LeaveType;
    reason: string;
    startDate: Date;
    endDate: Date;
    status: Status;
};